spl/src/metadata.rs

```diff
@@ -786,29 +805,25 @@ impl Deref for MetadataAccount {
 impl anchor_lang::IdlBuild for MetadataAccount {}
 
 #[derive(Clone, Debug, PartialEq)]
-pub struct MasterEditionAccount(mpl_token_metadata::state::MasterEditionV2);
-
-impl MasterEditionAccount {
-    pub const LEN: usize = mpl_token_metadata::state::MAX_MASTER_EDITION_LEN;
-}
+pub struct MasterEditionAccount(mpl_token_metadata::accounts::MasterEdition);
 
 impl anchor_lang::AccountDeserialize for MasterEditionAccount {
     fn try_deserialize(buf: &mut &[u8]) -> anchor_lang::Result<Self> {
         let me = Self::try_deserialize_unchecked(buf)?;
-        if me.key != mpl_token_metadata::state::MasterEditionV2::key() {
+        if me.key != mpl_token_metadata::types::Key::MasterEditionV2 {
```

Comment:
Are the changes for the `MasterEditionAccount` correct? The account changed from `MasterEditionV2` to `MasterEdition` but we are checking the key of `MasterEditionV2` here.

https://github.com/coral-xyz/anchor/blob/a465ae6fb49cc9058f4674ae40a82b91306ef0b2/spl/src/metadata.rs#L808

Reply:
Yes, the account struct `MasterEdition` is compatible with both `V1` and `v2`. And the test is equivalent – i.e., `MasterEditionV2::key()` gives the same value as `Key::MasterEditionV2`.